plugins/orchestrator-backend/src/service/SonataFlowService.ts

```diff
@@ -233,10 +235,18 @@ export class SonataFlowService {
     return undefined;
   }
 
-  public async fetchProcessInstances(): Promise<ProcessInstance[] | undefined> {
-    const graphQlQuery =
-      '{ ProcessInstances ( orderBy: { start: ASC }, where: {processId: {isNull: false} } ) { id, processName, processId, state, start, lastUpdate, end, nodes { id }, variables, parentProcessInstance {id, processName, businessKey} } }';
-
+  public async fetchProcessInstances(
```

Comment:
Could you please check if the pagination strategy defined in this PR is compatible with [this library](https://github.com/kiegroup/kie-tools/tree/main/packages/runtime-tools-gateway-api)?

Example for fetching instances:
- https://github.com/kiegroup/kie-tools/blob/main/packages/runtime-tools-gateway-api/src/graphql/types.tsx#L1525
- https://github.com/kiegroup/kie-tools/blob/main/packages/runtime-tools-gateway-api/src/gatewayApi/apis.tsx#L45

I'm saying this because ideally we will have to replace the code in this class with calls to [this library](https://github.com/kiegroup/kie-tools/tree/main/packages/runtime-tools-gateway-api) to avoid duplication.

Reply:
@caponetto so you're requesting to adopt kie-tools/runtime-tools-gateway-api package to perform these queries?
This PR seems compatible with it, as it uses the same entities and schemas, so it should not be an issue to update the implementation.

Reply:
I'm not requesting to adopt the new library now (we will need a new release first). I just want to make sure we are not diverging in the strategies because at some point the code will be replaced in favor of the library. If the implementation here is compatible, we are good to go :)